Formula/go.rb

```diff
@@ -54,6 +54,8 @@ def install
 
     cd "src" do
       ENV["GOROOT_FINAL"] = libexec
+      # Required for Apple Silicon Rosetta: Without it, Go will hang while building
+      ENV["GODEBUG"] = "asyncpreemptoff=1"
```

Comment:
We probably want to limit this… I don't think we have a way to switch for Rosetta only, so `if MacOS.version >= :big_sur` is probably good

Reply:
Apple has a way to detect rosetta, but I'm not familiar enough with Ruby to implement it myself:
https://developer.apple.com/documentation/apple_silicon/about_the_rosetta_translation_environment?language=objc

This might make sense as a general Homebrew helper, if you want to open that door.